src/integrators/linear_integrators.jl

```diff
@@ -131,17 +131,29 @@ function perform_step!(integrator,cache::MidpointSplittingCache,f=integrator.f)
 
   copy!(tmp, uprev)
   for B in reverse(Bs)
-    u .= expm((dt/2)*B)*tmp
+    propagate!(u, B, dt/2, tmp)
     @swap!(tmp,u)
   end
 
-  u .= expm(dt*A)*tmp
+  propagate!(u, A, dt, tmp)
 
   for B in Bs
-    tmp .= expm((dt/2)*B)*u
+    propagate!(tmp, B, dt/2, u)
     @swap!(u,tmp)
   end
 
   f(t+dt,u,integrator.fsallast)
   @pack integrator = t,dt,u
 end
+
+function propagate!(v, L, t, u)
```

Comment:
This should probably be part of the Operator's interface, not internal to the solver. So it should probably be in DiffEqBase and get tested there, and the solvers can just import the function and assume it works on whatever the given operator is.